packages/loader/container-loader/src/container.ts

```diff
@@ -791,31 +790,35 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             return;
         }
 
+        // If attachment blobs were uploaded in detached state we will go through a different attach flow
+        const hasAttachmentBlobs = this.loader.services.detachedBlobStorage !== undefined
+            && this.loader.services.detachedBlobStorage.size > 0;
+
         this.attachInProgress = true;
         try {
             assert(this.deltaManager.inbound.length === 0, 0x0d6 /* "Inbound queue should be empty when attaching" */);
             // Only take a summary if the container is in detached state, otherwise we could have local changes.
             // In failed attach call, we would already have a summary cached.
+            let summary: ISummaryTree;
             if (this._attachState === AttachState.Detached) {
```

Comment:
With removal of cachedAttachSummary, I believe this IF can be replaced with assertion, as there should be no way to call this API twice. In fact, it would be great to prevent it, but also mention in breaking.md as that's change in behavior that will bite clients unless they know about it and could adjust.